### PR TITLE
datasource images support snapshotSet

### PR DESCRIPTION
**Pull request by crab21**
$ make testacc TESTARGS=" -run=TestAccTencentCloudDataSourceImagesBase"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -run=TestAccTencentCloudDataSourceImagesBase -timeout 120m
?   	github.com/terraform-providers/terraform-provider-tencentcloud	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/terraform-providers/terraform-provider-tencentcloud/gendoc	(cached) [no tests to run]

=== RUN   TestAccTencentCloudDataSourceImagesBase
--- PASS: TestAccTencentCloudDataSourceImagesBase (13.88s)
PASS
ok  	github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud	13.912s
?   	github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud/connectivity	[no test files]
?   	github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud/internal/helper	[no test files]
?   	github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud/ratelimit	[no test files]

### CODE REVIEW

File: tencentcloud/data_source_tc_images.go [hunk 1]
```diff
@@ -270,3 +310,50 @@ func dataSourceTencentCloudImagesRead(d *schema.ResourceData, meta interface{})
 
 	return nil
 }
+
+func imagesReadSnapshotByIds(image *cvm.Image, cbsService CbsService, ctx context.Context) (snapshotResults []map[string]interface{}, errSnap error) {
```
Comment (Sherlock-Holo) [inline tencentcloud/data_source_tc_images.go:314]: 按照 google 的建议，`ctx context.Context` 放第一个参数

File: tencentcloud/service_tencentcloud_cbs.go [hunk 2]
```diff
@@ -226,6 +226,44 @@ func (me *CbsService) DescribeSnapshotById(ctx context.Context, snapshotId strin
...
+		if err != nil {
+			log.Printf("[CRITAL]%s api[%s] fail, request body [%s], reason[%s]\n",
+				logId, request.GetAction(), request.ToJsonString(), err.Error())
+			errRet = err
+			return
+		}
+		log.Printf("[DEBUG]%s api[%s] success, request body [%s], response body [%s]\n",
+			logId, request.GetAction(), request.ToJsonString(), response.ToJsonString())
+
+		if response == nil || len(response.Response.SnapshotSet) < 1 {
```
Comment (Sherlock-Holo) [inline tencentcloud/service_tencentcloud_cbs.go:253]: `err == nil` 时 `response` 不会为 nil

File: tencentcloud/data_source_tc_images.go [hunk 3]
```diff
@@ -136,6 +137,35 @@ func dataSourceTencentCloudImages() *schema.Resource {
 							Computed:    true,
 							Description: "Whether support cloud-init.",
 						},
+						"snapshots": {
```
Comment (brickzzhang) [inline tencentcloud/data_source_tc_images.go:140]: 一般list属性都写成_list ？

File: tencentcloud/service_tencentcloud_cbs.go [hunk 4]
```diff
@@ -226,6 +226,44 @@ func (me *CbsService) DescribeSnapshotById(ctx context.Context, snapshotId strin
...
+		return
+	}
+	request.SnapshotIds = snapshotIdsParam
+
+	offset := 0
+	pageSize := 100
+	for {
+		request.Offset = helper.IntUint64(offset)
+		request.Limit = helper.IntUint64(pageSize)
+		ratelimit.Check(request.GetAction())
```
Comment (brickzzhang) [inline tencentcloud/service_tencentcloud_cbs.go:242]: 加一下重试吧，如果某次请求失败，能重新请求

File: tencentcloud/data_source_tc_images.go (hunk 3 above)
Comment (crab21) [reply tencentcloud/data_source_tc_images.go:140]: > 一般list属性都写成_list ？

参考了上面的images

File: tencentcloud/service_tencentcloud_cbs.go [hunk 5]
```diff
@@ -226,6 +227,53 @@ func (me *CbsService) DescribeSnapshotById(ctx context.Context, snapshotId strin
...
+			return nil
+		})
+
+		if err != nil {
+			log.Printf("[CRITAL]%s api[%s] fail, request body [%s], reason[%s]\n",
+				logId, request.GetAction(), request.ToJsonString(), err.Error())
+			errRet = err
+			return
+		}
+		log.Printf("[DEBUG]%s api[%s] success, request body [%s], response body [%s]\n",
```
Comment (oliverpei) [inline tencentcloud/service_tencentcloud_cbs.go:261]: 这个日志可以删除，有统一的请求日志打印

File: tencentcloud/data_source_tc_images.go [hunk 6]
```diff
@@ -270,3 +309,32 @@ func dataSourceTencentCloudImagesRead(d *schema.ResourceData, meta interface{})
 
 	return nil
 }
+
+func imagesReadSnapshotByIds(ctx context.Context, cbsService CbsService, image *cvm.Image) (snapshotResults []map[string]interface{}, errRet error) {
```
Comment (Himer) [inline tencentcloud/data_source_tc_images.go:313]: 此功能建议写在CbsService中

File: website/docs/d/images.html.markdown [hunk 7]
```diff
@@ -46,6 +46,11 @@ In addition to all arguments above, the following attributes are exported:
   * `image_type` - Type of the image.
   * `os_name` - OS name of the image.
   * `platform` - Platform of the image.
+  * `snapshots` - List of snapshot details.
+    * `disk_size` - Size of the cloud disk used to create the snapshot; unit: GB.
+    * `disk_usage` - Type of the cloud disk used to create the snapshot.
+    * `snapshot_id` - Snapshot ID.
+    * `snapshot_name` - Snapshot name, the user-defined snapshot alias.
```
Comment (Himer) [inline website/docs/d/images.html.markdown:53]: 新增加特性需要修改changelog.md

File: tencentcloud/service_tencentcloud_cbs.go [hunk 8]
```diff
@@ -226,6 +227,50 @@ func (me *CbsService) DescribeSnapshotById(ctx context.Context, snapshotId strin
...
+		}
+
+		snapshots = append(snapshots, response.Response.SnapshotSet...)
+		if len(response.Response.SnapshotSet) < pageSize {
+			break
+		}
+		offset += pageSize
+	}
+	return
+}
```
Comment (Himer) [inline tencentcloud/service_tencentcloud_cbs.go:272]: func (me *CbsService) DescribeSnapshotByIds(ctx context.Context, snapshotIdsParam []*string) (snapshots []*cbs.Snapshot, errRet error) {
	if len(snapshotIdsParam) == 0 {
		return
	}
	var (
		logId = getLogId(ctx)
		request  = cbs.NewDescribeSnapshotsRequest()
		err      error
		response *cbs.DescribeSnapshotsResponse
		offset, pageSize uint64 = 0, 100
	)
	request.SnapshotIds = snapshotIdsParam
	request.Offset = &offset
	request.Limit = &pageSize

	for {
		err = resource.Retry(readRetryTimeout, func() *resource.RetryError {
			ratelimit.Check(request.GetAction())
			response, err = me.client.UseCbsClient().DescribeSnapshots(request)
			if err != nil {
				return retryError(err, InternalError)
			}
			return nil
		})
		if err != nil {
			log.Printf("[CRITAL]%s api[%s] fail, request body [%s], reason[%s]\n",
				logId, request.GetAction(), request.ToJsonString(), err.Error())
			errRet = err
			return
		}
		snapshots = append(snapshots, response.Response.SnapshotSet...)
		if len(response.Response.SnapshotSet) < pageSize {
			break
		}
		offset += pageSize
	}
	return
}

File: tencentcloud/service_tencentcloud_cbs.go [hunk 9]
```diff
@@ -226,6 +227,52 @@ func (me *CbsService) DescribeSnapshotById(ctx context.Context, snapshotId strin
...
+			return nil
+		})
+
+		if err != nil {
+			log.Printf("[CRITAL]%s api[%s] fail, request body [%s], reason[%s]\n",
+				logId, request.GetAction(), request.ToJsonString(), err.Error())
+			errRet = err
+			return
+		}
+		if len(response.Response.SnapshotSet) < 1 {
```
Comment (Himer) [inline tencentcloud/service_tencentcloud_cbs.go:263]: 去掉
if len(response.Response.SnapshotSet) < 1 {
			break
		}

**Pull request merged**